File: API/DTOs/PlantDto.cs
using API.Models;
using System.ComponentModel.DataAnnotations;

namespace API.DTOs
{
    public class PlantResponseDto
    {
        public int PlantId { get; set; }
        public int UserId { get; set; }
        public int? DeviceId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Species { get; set; }
        public decimal? OptimalMoistureMin { get; set; }
        public decimal? OptimalMoistureMax { get; set; }
        public decimal? OptimalLightMin { get; set; }
        public decimal? OptimalLightMax { get; set; }
        public string? Notes { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool IsActive { get; set; }
    }

    public class CreatePlantDto
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        public int? DeviceId { get; set; }

        [MaxLength(100)]
        public string? Species { get; set; }

        public decimal? OptimalMoistureMin { get; set; }
        public decimal? OptimalMoistureMax { get; set; }
        public decimal? OptimalLightMin { get; set; }
        public decimal? OptimalLightMax { get; set; }
        public string? Notes { get; set; }
    }

    public class UpdatePlantDto
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        public int? DeviceId { get; set; }

        [MaxLength(100)]
        public string? Species { get; set; }

        public decimal? OptimalMoistureMin { get; set; }
        public decimal? OptimalMoistureMax { get; set; }
        public decimal? OptimalLightMin { get; set; }
        public decimal? OptimalLightMax { get; set; }
        public string? Notes { get; set; }
        public bool IsActive { get; set; }
    }
}

File: Mobile/composeApp/src/androidMain/kotlin/com/plantiq/data/model/PlantDto.kt
package com.plantiq.data.model

import com.google.gson.annotations.SerializedName

data class PlantResponseDto(
    @SerializedName("plantId") val plantId: Int,
    @SerializedName("userId") val userId: Int,
    @SerializedName("deviceId") val deviceId: Int?,
    @SerializedName("name") val name: String,
    @SerializedName("species") val species: String?,
    @SerializedName("optimalMoistureMin") val optimalMoistureMin: Double?,
    @SerializedName("optimalMoistureMax") val optimalMoistureMax: Double?,
    @SerializedName("optimalLightMin") val optimalLightMin: Double?,
    @SerializedName("optimalLightMax") val optimalLightMax: Double?,
    @SerializedName("notes") val notes: String?,
    @SerializedName("isActive") val isActive: Boolean,
    @SerializedName("wateringMode") val wateringMode: String
)

data class CreatePlantDto(
    @SerializedName("name") val name: String,
    @SerializedName("deviceId") val deviceId: Int?,
    @SerializedName("species") val species: String?,
    @SerializedName("optimalMoistureMin") val optimalMoistureMin: Double?,
    @SerializedName("optimalMoistureMax") val optimalMoistureMax: Double?,
    @SerializedName("optimalLightMin") val optimalLightMin: Double?,
    @SerializedName("optimalLightMax") val optimalLightMax: Double?,
    @SerializedName("notes") val notes: String?,
    @SerializedName("wateringMode") val wateringMode: String = "Manual"
)

data class UpdatePlantDto(
    @SerializedName("name") val name: String,
    @SerializedName("deviceId") val deviceId: Int?,
    @SerializedName("species") val species: String?,
    @SerializedName("optimalMoistureMin") val optimalMoistureMin: Double?,
    @SerializedName("optimalMoistureMax") val optimalMoistureMax: Double?,
    @SerializedName("optimalLightMin") val optimalLightMin: Double?,
    @SerializedName("optimalLightMax") val optimalLightMax: Double?,
    @SerializedName("notes") val notes: String?,
    @SerializedName("isActive") val isActive: Boolean,
    @SerializedName("wateringMode") val wateringMode: String
)

data class PlantSpeciesResponseDto(
    @SerializedName("speciesId") val speciesId: Int,
    @SerializedName("name") val name: String,
    @SerializedName("defaultMoistureMin") val defaultMoistureMin: Double?,
    @SerializedName("defaultMoistureMax") val defaultMoistureMax: Double?,
    @SerializedName("defaultLightMin") val defaultLightMin: Double?,
    @SerializedName("defaultLightMax") val defaultLightMax: Double?
)

data class CreatePlantSpeciesDto(
    @SerializedName("name") val name: String,
    @SerializedName("defaultMoistureMin") val moistureMin: Double?,
    @SerializedName("defaultMoistureMax") val moistureMax: Double?,
    @SerializedName("defaultLightMin") val lightMin: Double?,
    @SerializedName("defaultLightMax") val lightMax: Double?
)
